### PR TITLE
Improve CreateEffectWarhead support for alpha/reflection/shadow

### PR DESCRIPTION
**Pull request by reaperrr**
Adds `ForceDisplayAtGroundLevel` aiming at making it easier for modders to add alpha glow effects, water reflections, shadows etc. to explosions.

Not really applicable to TS yet due to #13430, unfortunately.

### CODE REVIEW

**Comment by GraionDilach** [comment]
I don't think you need all that duplication stuff - a second CEW can be used for such already.

**Comment by reaperrr** [comment]
Updated.

**Review by GraionDilach — APPROVED** [review]
:+1:

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs [hunk 1]
```diff
@@ -31,6 +31,9 @@ public class CreateEffectWarhead : Warhead, IRulesetLoaded<WeaponInfo>
 		[Desc("Remap explosion effect to player color, if art supports it.")]
 		public readonly bool UsePlayerPalette = false;
 
+		[Desc("Display explosion effect at ground level, regardless of explosion altitude.")]
+		public readonly bool ForceGroundLevel = true;
```
Comment (pchote) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: Where is this used?

File: OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs [hunk 2]
```diff
@@ -117,9 +120,12 @@ public override void DoImpact(Target target, Actor firedBy, IEnumerable<int> dam
 			if (UsePlayerPalette)
 				palette += firedBy.Owner.InternalName;
 
+			var dat = world.Map.DistanceAboveTerrain(pos);
+			var groundPos = new WPos(pos.X, pos.Y, pos.Z - dat.Length);
```
Comment (rob-v) [inline OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs]: This could be calculated only when ForceGroundLevel .

**Comment by reaperrr** [comment]
Updated.

**Comment by reaperrr** [comment]
@atlimit8 I already updated the code accordingly.

**Review by pchote — APPROVED** [review]
LGTM :+1:

I'm holding off merging this because I don't know whether `ForceGroundLevel` conflicts with the more recent changes/thoughts in #13543.  If it doesn't, then please go ahead and merge this.

**Comment by reaperrr** [comment]
> I'm holding off merging this because I don't know whether ForceGroundLevel conflicts with the more recent changes/thoughts in #13543. If it doesn't, then please go ahead and merge this.

Not conceptually (imo), but the field name is a bit ambigous in that regard, so I've renamed it to `ForceDisplayAtGroundLevel`.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/63985f83ee8fdbab57797cd30c213efd19b51430...269c50d6f48868b15f1d303feb974aba97d7f956

**Pull request merged**